upgrade/steps.go

```diff
@@ -339,116 +338,69 @@ func InformGitHub(
 //
 // This is predicated on updating to the latest version being safe. We will need to
 // revisit this when a new major version of the plugin SDK is released.
-func getLatestTFPluginSDKReplace(ctx context.Context, repo ProviderRepo) (step.Step, *bool) {
-	name := "Update TF Plugin SDK Fork"
-	stepFail := func(msg string, a ...any) step.Step {
-		return step.F(name, func() (string, error) {
-			return "", fmt.Errorf(msg, a...)
-		})
-	}
-
-	didReplace := new(bool)
-
+func getLatestTFPluginSDKReplace(ctx context.Context, repo ProviderRepo, targetSHA *string) step.Step {
 	// We do discover in a step.Computed so if the fork isn't present, it isn't
 	// displayed to the user.
-	return step.Computed(func() step.Step {
-		goModFile, err := os.ReadFile("go.mod")
+	return step.F("Update TF Plugin SDK Fork", func() (string, error) {
+		// If the fork is present, we need to figure out the SHA of the
+		// latest upstream version to use.
+		const hostRepo = "https://github.com/pulumi/terraform-plugin-sdk.git"
+		result, err := exec.CommandContext(ctx, "git",
+			"ls-remote", "--heads", hostRepo).Output()
 		if err != nil {
-			return stepFail("could not fine go.mod: %w", err)
+			return "", fmt.Errorf("could not get branches: %w", err)
 		}
-		goMod, err := modfile.Parse("go.mod", goModFile, nil)
-		if err != nil {
-			return stepFail("could not parse go.mod: %w", err)
-		}
-
-		const targetSrc = "github.com/hashicorp/terraform-plugin-sdk/v2"
-
-		var require *modfile.Require
-		for _, r := range goMod.Require {
-			if r.Mod.Path == targetSrc {
-				require = r
+		lines := strings.Split(string(result), "\n")
+		versions := make([]*semver.Version, len(lines))
+		shas := make([]string, len(lines))
+		highest := -1
+		for i, line := range lines {
+			split := strings.Split(strings.TrimSpace(line), "\t")
+			if len(split) < 2 {
+				continue
+			}
+			shas[i] = split[0]
```

Comment:
do we need shas? I'm not seeing where they're being used, but perhaps I'm mistaken?

Reply:
Unfortunately, we do. `replace` directives don't take "version queries", they take versions. Since the release we are targeting is the head of the latest branch, we need to insert a `replace` statement pointing to the SHA of that branch HEAD.